dataflow/src/main/java/org/checkerframework/dataflow/analysis/FlowExpressions.java

```diff
@@ -1057,6 +1065,190 @@ public String toString() {
         }
     }
 
+    /** FlowExpression.Receiver for binary operations. */
+    public static class BinaryOperation extends Receiver {
+
+        /** The binary operation kind. */
+        protected final Kind operationKind;
+        /** Receiver of the left operand. */
+        protected final Receiver left;
+        /** Receiver of the right operand. */
+        protected final Receiver right;
+
+        /**
+         * @param node the binary operation node.
+         * @param left the left operand receiver.
+         * @param right the right operand receiver.
+         */
+        public BinaryOperation(BinaryOperationNode node, Receiver left, Receiver right) {
+            super(node.getType());
+            this.operationKind = node.getTree().getKind();
+            this.left = left;
+            this.right = right;
+        }
+
+        /** @return the binary operation kind. */
+        public Kind getOperationKind() {
+            return operationKind;
+        }
+
+        /** @return the left receiver. */
+        public Receiver getLeft() {
+            return left;
+        }
+
+        /** @return the right receiver. */
+        public Receiver getRight() {
+            return right;
+        }
+
+        /** @return true if the binary operation is commutative. */
+        private boolean isCommutative() {
+            switch (operationKind) {
+                case PLUS:
+                case MULTIPLY:
+                case AND:
+                case OR:
+                case XOR:
+                case EQUAL_TO:
+                case NOT_EQUAL_TO:
+                case CONDITIONAL_AND:
+                case CONDITIONAL_OR:
+                    return true;
+                default:
+                    return false;
+            }
+        }
+
+        @Override
+        public boolean containsOfClass(Class<? extends Receiver> clazz) {
+            if (getClass() == clazz) {
+                return true;
+            }
+            return right.containsOfClass(clazz) || left.containsOfClass(clazz);
```

Comment:
Nit: can you change these and other comparisons to have `left` on the left of `right`? It just looks odd to me to compare `right` before `left`. It doesn't matter here, but the evaluation order would be `left` and then `right`.